agent/src/beerocks/slave/son_slave_thread.cpp

```diff
@@ -4432,6 +4432,52 @@ bool slave_thread::handle_channel_preference_query(Socket *sd, ieee1905_1::CmduM
     return message_com::send_cmdu(ap_manager_socket, cmdu_tx);
 }
 
+/**
+ * @brief check if channel is restricted in a given preference
+ * 
+ * @param channel channel to check
+ * @param preference channel preference
+ * @return true if channel is restricted
+ * @return false if channel is not restricted
+ */
+static bool is_channel_restricted(const beerocks::message::sWifiChannel channel,
```

Comment:
(nitpick) it would make more sense to return the preference value instead, or NON_OPERABLE.